python/pyspark/sql/tests/udf_type_tests/test_udf_input_types.py

```diff
@@ -115,7 +115,34 @@ def value_udf(x):
                     return x
 
                 def value_str(x):
-                    return str(x)
+                    class NpPrintable:
+                        def __init__(self, x):
+                            self.x = x
+
+                        def __repr__(self):
+                            return f"np.{self.x.dtype}({self.x.item()})"
+
+                    # Numpy 1.x __repr__ returns a different format. We only care about
+                    # types and values of the elements,
+                    # so we accept this difference and implement our own repr to make
+                    # tests with numpy 1 return the same format as numpy 2.
+                    def convert_to_numpy_printable(x):
+                        if isinstance(x, Row):
+                            return Row(
+                                **{k: convert_to_numpy_printable(v) for k, v in x.asDict().items()}
```

Comment:
Does this always preserve column order?

Reply:
technically yes, since 3.7 python dicts preserve insertion order. I rewrote this to just operate direclty on tuples to make it clear

Reply:
Makes sense, thanks for adjusting. I think asDict is better too but it depends on your preference